File: duAn1/duAn1/wwwroot/js/price-formatter.js
// Format giá tiền theo kiểu VND
// Ví dụ: 1000000 -> "1.000.000 VNĐ"
function formatVND(price) {
    if (!price || price === 0) {
        return '0 VNĐ';
    }
    
    // Chuyển thành số nguyên nếu là chuỗi
    const numPrice = typeof price === 'string' ? parseInt(price) : price;
    
    // Format với dấu chấm phân cách hàng ngàn
    return numPrice.toLocaleString('vi-VN') + ' VNĐ';
}

// Hàm này có thể được gọi sau khi DOM thay đổi để update tất cả giá đã format sẵn
function updatePriceDisplay(selector, priceValue) {
    const element = document.querySelector(selector);
    if (element) {
        element.textContent = formatVND(priceValue);
    }
}

File: duAn1/duAn1/Utils/PriceFormatter.cs
using System;

namespace duAn1.Utils
{
    public static class PriceFormatter
    {
        /// <summary>
        /// Format giá tiền theo định dạng VND chuẩn
        /// Ví dụ: 1000000 -> "1.000.000 VNĐ"
        /// </summary>
        public static string FormatVND(int? price)
        {
            if (price == null || price == 0)
                return "0 VNĐ";

            return price.Value.ToString("N0") + " VNĐ";
        }

        /// <summary>
        /// Format giá tiền theo định dạng VND chuẩn
        /// Ví dụ: 1000000 -> "1.000.000 VNĐ"
        /// </summary>
        public static string FormatVND(long? price)
        {
            if (price == null || price == 0)
                return "0 VNĐ";

            return price.Value.ToString("N0") + " VNĐ";
        }

        /// <summary>
        /// Format giá tiền theo định dạng VND chuẩn
        /// Ví dụ: 1000000 -> "1.000.000 VNĐ"
        /// </summary>
        public static string FormatVND(decimal? price)
        {
            if (price == null || price == 0)
                return "0 VNĐ";

            return ((long)price.Value).ToString("N0") + " VNĐ";
        }
    }
}
